docs/DeveloperGuide.md

```diff
@@ -1,29 +1,342 @@
+    <br/>
+    <i>Figure 3: UML diagram for the Expenditure Categories component</i>
+</p>
+
+In the diagram, the aforementioned expenditure categories inherit from the `Expenditure` class. The `ExpenditureList` class is a composition of expenditures of `Expenditure` type. 
+
+`Expenditure` has a multiplicity of `*` to `ExpenditureList` as an empty expenditure list is instantiated at the beginning of the program, and any number of expenditures can be added to the expenditure list. Thus, it is also observed that the `ExpenditureList` class is an *composition* of `Expenditure`.
+
+### 3.4. Command Component
+
+The `Command` component is represented by the `command` package. The `command` package contains all the available user commands supported by the application. These commands are utilised by the user to interact with the expenditure types and the expenditure list. 
+
+The `AcademicExpenditureCommand`, `AccommodationExpenditureCommand`, `BorrowExpenditureCommand`, `EntertainmentExpenditureCommand`, `FoodExpenditureCommand`, `LendExpenditureCommand`, `OtherExpenditureCommand`, `TransportExpenditureCommand`, `TuitionExpenditureCommand` commands contain the operations pertaining to adding a new expenditure into the list of expenditures.
+
+The `CheckBudgetCommand` class class contains the operations pertaining to comparing the total expenditure amount with a budget set by the user. The budget is set with the `SetBudgetCommand`.
+
+The `DeleteCommand` class contains the operations pertaining to deleting a specific expenditure from the list of expenditures.
+
+The `DuplicateCommand` class contains the operations pertaining to duplicating a specific expenditure from the list of expenditures.
+
+The `EditCommand` class contains the operations pertaining to editing a expenditure from the list of expenditures.
+
+The `ExitCommand` class contains the operation that safely closes the application.
+
+The `FindCommand`class contains the operations pertaining to searching the list of expenditures for expenditures that match the keyword entered by the user.
+
+The `HelpCommand` class contains the operation pertaining to providing the user a user interface to the instructions on the use of the application.
+
+The `InvalidCommand` class is instantiated when an unrecognised command is entered by the user.
+
+The `SetBudgetCommand` class contains the operations in setting an amount of money users would like to budget.
+
+The `SortCommand` class contains the operations pertaining to sorting the list of expenditures by amount or by date.
+
+The `ViewDateExpenditureCommand` and `ViewTypeExpenditureCommand` classes contain the operations pertaining to displaying a filtered expenditure list by the expenditure date and type respectively.
+
+Below shows the UML diagram representing the `command` package.
+<p align="center">
+    <img src="team/images/umlCommandClassDiagram.png" width="100%">
+    <br/>
+    <i>Figure 4: UML diagram for the command package</i>
+</p>
+
```

Comment:
Perhaps you could split the UML diagram into different parts.  The UML diagram is too small and unreadable.